dash0_spam_filter: 404 on Read when the filter was deleted out-of-band blocks plan/apply

## Summary

If a `dash0_spam_filter` resource is removed out-of-band (for example, deleted via the Dash0 web app) after Terraform created it, the next `terraform plan` or `apply` fails instead of detecting the drift and offering to recreate the resource.

```
│ Error: Client Error
│
│   with module.dash0.dash0_spam_filter.spam_filters["health.yaml.tftpl"],
│   on dash0/rules.tf line 24, in resource "dash0_spam_filter" "spam_filters":
│   24: resource "dash0_spam_filter" "spam_filters" {
│
│ Unable to read spam filter, got error: dash0 api error: spam filter not
│ found (status: 404, trace_id: 50fdd281f364acd93af01d04049fb89a)
```

Terraform's contract for a resource that no longer exists server-side is to remove it from state during `Read` so the next plan proposes to recreate it — not to surface a hard error. Today the user has to run `terraform state rm` manually to unblock every affected resource.

## Root cause

`SpamFilterResource.Read` (`internal/provider/spam_filter_resource.go`) forwards any error from `GetSpamFilter` straight into a generic `"Client Error"` diagnostic:

```go
apiResponseJSON, err := r.client.GetSpamFilter(ctx, state.Origin.ValueString(), state.Dataset.ValueString())
if err != nil {
    resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to read spam filter, got error: %s", err))
    return
}
```

It never checks `dash0.IsNotFound(err)` and never calls `resp.State.RemoveResource(ctx)`.

`TeamResource.Read` (`internal/provider/team_resource.go:315-328`) already implements the correct pattern for this exact situation:

```go
if err != nil {
    if dash0.IsNotFound(err) {
        tflog.Debug(ctx, fmt.Sprintf("Team %s no longer exists on the server; removing from state", state.Origin.ValueString()))
        resp.State.RemoveResource(ctx)
        return
    }
    resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to read team, got error: %s", err))
    return
}
```

`TeamResource` is also the only resource in the provider with this handling — the same gap exists in `check_rule_resource.go`, `dashboard_resource.go`, `notification_channel_resource.go`, `recording_rule_resource.go`, `synthetic_check_resource.go`, and `view_resource.go`. Filing this issue against `dash0_spam_filter` since that's what was reported in practice, but a fix should likely apply the same `dash0.IsNotFound` pattern to `Delete` (idempotent destroy, per `TeamResource.Delete`) across all of these resource types.

## Reproduction

Branch [`repro/spam-filter-404-on-read`](https://github.com/dash0hq/terraform-provider-dash0/tree/repro/spam-filter-404-on-read) adds a unit test, `TestSpamFilterResource_Read_404IsNotHandledAsGone` in `internal/provider/spam_filter_resource_test.go`, that mocks `GetSpamFilter` returning a genuine `dash0.APIError{StatusCode: 404}` and asserts the behavior `TeamResource.Read` already has (no diagnostics error, state removed). It currently fails against `main`:

```
go test ./internal/provider/... -run TestSpamFilterResource_Read_404IsNotHandledAsGone -v
=== RUN   TestSpamFilterResource_Read_404IsNotHandledAsGone
    spam_filter_resource_test.go:81:
        Error:      Should be false
        Messages:   a 404 from GetSpamFilter should be treated as 'resource is gone', not surfaced as a hard error
--- FAIL: TestSpamFilterResource_Read_404IsNotHandledAsGone (0.00s)
```

## Expected behavior

`Read` detects `dash0.IsNotFound(err)` and calls `resp.State.RemoveResource(ctx)`, matching `TeamResource`. `terraform plan` then proposes recreating the spam filter instead of erroring out.

--- a/.chloggen/repro-spam-filter-404-on-read.yaml
+++ b/.chloggen/repro-spam-filter-404-on-read.yaml
@@ -1,0 +1,26 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern (e.g. dashboards, check_rules, views)
+component: spam_filters
+
+# A brief description of the change. Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: "Treat a 404 from `Read` as the spam filter no longer existing, removing it from state instead of failing `plan`/`apply`."
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: [164]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:
+
+# If your change doesn't affect end users or the exported elements of any package,
+# you should instead start your pull request title with "chore" or use the "Skip Changelog" label.
+# Optional: The change log or logs in which this entry should be included.
+# e.g. '[user]' or '[user, api]'
+# Include 'user' if the change is relevant to end users.
+# Default: '[user]'
+change_logs: []

--- a/internal/provider/spam_filter_resource.go
+++ b/internal/provider/spam_filter_resource.go
@@ -9,6 +9,7 @@ import (
 	"github.com/hashicorp/terraform-plugin-framework/diag"
 	"github.com/hashicorp/terraform-plugin-framework/path"
 
+	dash0 "github.com/dash0hq/dash0-api-client-go"
 	"github.com/dash0hq/terraform-provider-dash0/internal/converter"
 	"github.com/dash0hq/terraform-provider-dash0/internal/provider/client"
 	customplanmodifier "github.com/dash0hq/terraform-provider-dash0/internal/provider/planmodifier"
@@ -185,6 +186,11 @@ func (r *SpamFilterResource) Read(ctx context.Context, req resource.ReadRequest,
 
 	apiResponseJSON, err := r.client.GetSpamFilter(ctx, state.Origin.ValueString(), state.Dataset.ValueString())
 	if err != nil {
+		if dash0.IsNotFound(err) {
+			tflog.Debug(ctx, fmt.Sprintf("Spam filter %s no longer exists on the server; removing from state", state.Origin.ValueString()))
+			resp.State.RemoveResource(ctx)
+			return
+		}
 		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to read spam filter, got error: %s", err))
 		return
 	}

--- a/internal/provider/spam_filter_resource_test.go
+++ b/internal/provider/spam_filter_resource_test.go
@@ -4,12 +4,86 @@ import (
 	"context"
 	"testing"
 
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
 	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
 	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
 
+	dash0 "github.com/dash0hq/dash0-api-client-go"
 	customplanmodifier "github.com/dash0hq/terraform-provider-dash0/internal/provider/planmodifier"
 )
+
+// TestSpamFilterResource_Read_404IsNotHandledAsGone covers the bug reported in
+// https://github.com/dash0hq/terraform-provider-dash0/issues/164 (spam filter
+// deleted out-of-band via the UI, then a Terraform plan/apply runs Read).
+//
+// SpamFilterResource.Read now checks dash0.IsNotFound(err) and calls
+// resp.State.RemoveResource(ctx), so Terraform treats the resource as gone and
+// proposes to re-create it on the next apply, instead of surfacing a hard
+// error that blocks `terraform plan`/`apply` until the user manually runs
+// `terraform state rm`.
+func TestSpamFilterResource_Read_404IsNotHandledAsGone(t *testing.T) {
+	mockClient := &MockClient{}
+	r := &SpamFilterResource{client: mockClient}
+
+	notFound := &dash0.APIError{
+		StatusCode: 404,
+		Status:     "404 Not Found",
+		Message:    "spam filter not found",
+		TraceID:    "repro-trace-id",
+	}
+	mockClient.On("GetSpamFilter", mock.Anything, "tf_origin", "dataset-1").
+		Return("", notFound)
+
+	state := tfsdk.State{
+		Raw: tftypes.NewValue(
+			tftypes.Object{
+				AttributeTypes: map[string]tftypes.Type{
+					"origin":           tftypes.String,
+					"id":               tftypes.String,
+					"dataset":          tftypes.String,
+					"spam_filter_yaml": tftypes.String,
+				},
+			},
+			map[string]tftypes.Value{
+				"origin":           tftypes.NewValue(tftypes.String, "tf_origin"),
+				"id":               tftypes.NewValue(tftypes.String, nil),
+				"dataset":          tftypes.NewValue(tftypes.String, "dataset-1"),
+				"spam_filter_yaml": tftypes.NewValue(tftypes.String, "test-yaml"),
+			},
+		),
+		Schema: schema.Schema{
+			Attributes: map[string]schema.Attribute{
+				"origin":           schema.StringAttribute{Computed: true},
+				"id":               schema.StringAttribute{Computed: true},
+				"dataset":          schema.StringAttribute{Required: true},
+				"spam_filter_yaml": schema.StringAttribute{Required: true},
+			},
+		},
+	}
+
+	// The real terraform-plugin-framework runtime always seeds
+	// ReadResponse.State from the request state (schema included) before
+	// calling Resource.Read — see internal/fwserver/server_readresource.go.
+	// Mirror that here so RemoveResource(ctx), which needs the schema to
+	// build the null value, has one to work with.
+	req := resource.ReadRequest{State: state}
+	resp := &resource.ReadResponse{State: state}
+
+	r.Read(context.Background(), req, resp)
+
+	// What SHOULD happen, matching team_resource.go's dash0.IsNotFound branch:
+	assert.False(t, resp.Diagnostics.HasError(),
+		"a 404 from GetSpamFilter should be treated as 'resource is gone', not surfaced as a hard error")
+	assert.True(t, resp.State.Raw.IsNull(),
+		"state should be removed via RemoveResource so terraform plan can recreate the spam filter")
+
+	mockClient.AssertExpectations(t)
+}
 
 // TestSpamFilterResource_SharingAnnotationIgnored verifies that spam filters
 // do NOT preserve dash0.com/sharing — changes to it should not trigger a replan.

